### PR TITLE
Use IsGhost() to access ghost state

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Added set and clear ghost to further prevent direct access to flags

### CODE REVIEW

File: src/openrct2/paint/tile_element/Paint.Entrance.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ static void ride_entrance_exit_paint(paint_session* session, uint8_t direction,
         paint_util_push_tunnel_left(session, height, TUNNEL_6);
     }
 
-    if (!is_exit && !(tile_element->flags & TILE_ELEMENT_FLAG_GHOST) && tile_element->AsEntrance()->GetRideIndex() != 0xFF
+    if (!is_exit && !(tile_element->IsGhost()) && tile_element->AsEntrance()->GetRideIndex() != 0xFF
```
Comment (Gymnasiast) [inline src/openrct2/paint/tile_element/Paint.Entrance.cpp]:
```suggestion
    if (!is_exit && !(tile_element->IsGhost()) && tile_element->AsEntrance()->GetRideIndex() != RIDE_ID_NULL
```

File: src/openrct2/paint/tile_element/Paint.Entrance.cpp [hunk 2]
```diff
@@ -157,7 +157,7 @@ static void ride_entrance_exit_paint(paint_session* session, uint8_t direction,
...
         && stationObj->ScrollingMode != 0xFF)
```
Comment (Gymnasiast) [inline src/openrct2/paint/tile_element/Paint.Entrance.cpp]:
```suggestion
        && stationObj->ScrollingMode != SCROLLING_MODE_NONE)
```

**Comment by ZehMatt** [comment]
Is there a reason we use two functions as in, Set/Clear. I would rather see SetGhost(bool state) imho.

**Comment by duncanspumpkin** [comment]
I thought of having just one but didn't like the inefficiency of testing the bool. Besides it's much clearer setghost/clearghost than setghost true/false

**Comment by ZehMatt** [comment]
Most systems I have worked with usually just have basic Get/Set rather than Get/Set/Clear but thats way too trivial to argue about, its just my opinion, but I also see your point of checking before applying.

**Comment by Gymnasiast** [comment]
The thing is that all tile element methods that I introduced use @ZedMatt's approach. I think it's more important to be consistent.

**Comment by duncanspumpkin** [comment]
Okay will change it

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 3]
```diff
@@ -2185,7 +2185,7 @@ static void window_tile_inspector_scrollpaint(rct_window* w, rct_drawpixelinfo*
 
         const int32_t baseHeight = tileElement->base_height;
         const int32_t clearanceHeight = tileElement->clearance_height;
-        const bool ghost = (tileElement->flags & TILE_ELEMENT_FLAG_GHOST) != 0;
+        const bool ghost = (tileElement->IsGhost()) != 0;
```
Comment (jensj12) [inline src/openrct2-ui/windows/TileInspector.cpp]: No need for `!= 0` anymore here.

**Pull request merged**